Code/GraphMol/FileParsers/MolSGroupParsing.cpp

```diff
@@ -14,6 +14,35 @@
 
 #include <RDGeneral/FileParseException.h>
 
+#define PARSE_SGROUP_FIELD_OR_FAIL(sgroup, ParseSGroupFunc, ...) \
```

Comment:
I'm willing to go along with this if others think this is a good idea, but I think introducing macros into the code is awful.
Particularly macros which use `__VA_ARGS__`
These make the code harder to read and debug.

Can't you just be using functions for this stuff?

Reply:
I know, I know, macros are evil. I'll remove them, they are by no means necessary. I personally think they actually make that boilerplate code more readable, but that's probably because I'm the one who wrote the code.

Reply:
If they can't be replaced with functions and removing them is going to result in a ton of duplicated code, then definitely keep them. But otherwise, if they can be removed I'd appreciate it.